gramps/gen/utils/parents.py

```diff
@@ -0,0 +1,88 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2017
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+"""
+Utilities to get labels for parents
+"""
+#-------------------------------------------------------------------------
+#
+# Gramps modules
+#
+#-------------------------------------------------------------------------
+from gramps.gen.const import GRAMPS_LOCALE as glocale
+_ = glocale.translation.gettext
+from gramps.gen.lib.person import Person
+from gramps.gen.relationship import get_relationship_calculator
+
+# _T_ is a gramps-defined keyword -- see po/update_po.py and po/genpot.sh
+def _T_(value):
+    """ enable deferred translations (see Python docs 22.1.3.4) """
+    return value
+
+#-------------------------------------------------------------------------
+#
+# get labels for parents
+#
+#-------------------------------------------------------------------------
+def parents(db, family, xlocale):
+    """
+    Get the label for parent
+    """
+    if family.get_father_handle(): #[0]
+        father = db.get_person_from_handle(family.get_father_handle())
+        if family.get_mother_handle():
+            mother = db.get_person_from_handle(family.get_mother_handle())
+        else:
+            if father.gender == Person.FEMALE:
+                return [_T_("Woman"), _T_("Person")]
+            else:
+                return [_T_("Man"), _T_("Person")]
+    else:
+        if family.get_mother_handle(): #[1]
+            mother = db.get_person_from_handle(family.get_mother_handle())
+            if mother.gender == Person.MALE:
+                return [_T_("Person"), _T_("Man")]
+        else:
+            return [_T_("Person"), _T_("Person")]
+
+    # need it for above strings and as fallback
+    rel_father = _("Man")
+    rel_mother = _("Woman")
+
+    if len(family.get_child_ref_list()) > 0:
```

Comment:
We only get here if the family has two parents.  Do we want to describe a single parent as "Father" or "Mother"?